src/node_buffer.cc

```diff
@@ -730,12 +730,20 @@ void ReadDoubleBE(const FunctionCallbackInfo<Value>& args) {
 
 
 template <typename T, enum Endianness endianness>
-uint32_t WriteFloatGeneric(const FunctionCallbackInfo<Value>& args) {
+void WriteFloatGeneric(const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+
+  bool should_assert = !args[3]->IsTrue();
+  if (should_assert) {
+    THROW_AND_RETURN_UNLESS_BUFFER(env, args[0]);
+  }
   SPREAD_ARG(args[0], ts_obj);
 
-  T val = args[1]->NumberValue();
-  uint32_t offset = args[2]->Uint32Value();
-  CHECK_LE(offset + sizeof(T), ts_obj_length);
+  T val = args[1]->NumberValue(env->context()).FromMaybe(0);
```

Comment:
Why does this function take the `T` parameter? `T` is used as the return type for `v8::Value::NumberValue::FromMaybe`, which returns a `double`.